helix-core/src/main/java/org/apache/helix/common/caches/CustomizedStateCache.java

```diff
@@ -46,14 +46,9 @@ public CustomizedStateCache(ControlContextProvider contextProvider) {
 
   @Override
   protected Set<PropertyKey> PopulateParticipantKeys(HelixDataAccessor accessor,
-      Map<String, LiveInstance> liveInstanceMap) {
+      Map<String, LiveInstance> liveInstanceMap, Set<String> restrictedKeys) {
```

Comment:
I think we discussed to put the restrictedKeys to the CustomizedStateCache constructor as a final field. So you don't need to add this to these methods? The downside of the current design is that the current state cache will have to keep the unnecessary input.